web/frontend/src/lib/schedule.ts

```diff
@@ -0,0 +1,28 @@
+export const scheduleAsyncPoll = (callback: () => Promise<void>, interval: number) => {
+  let cancelled = false;
+  let timeoutId = -1;
```

Comment:
IIRC the typescript types allow `undefined` for `clearTimeout` (and it's certainly allowed at runtime) so I softly prefer that to a `-1` magic value for "not set"